src/components/RemountOnThemeChange.tsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useRef, useState } from "react"
+import { useAtom } from "jotai"
+
+import { $theme } from "../store/theme"
+
+interface Props {
+  children: React.ReactNode
+}
+
+/**
+ * WORKAROUND: remount the children when the theme changes.
```

Comment:
The MetabaseProvider has no way to know that the user session has changed, so this helps to avoid a full-page reload.